app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, optional: true, foreign_key: :users_id
+  has_many :likes, foreign_key: :posts_id
+  has_many :comments, foreign_key: :posts_id
```

Comment:
- [ ] Kindly remove all of these foreign_keys from your associations as they're not valid in this case. Instead, please specify which table do you want it to be related to and how?
1. a `Post` belongs_to a `User`, and the `User` has_many `Posts` so let's determine how do we want them to relate to each other, and in this case, we want to call the `User` with the `Author` name, so when we set the association in the `Post` model, we will say, well! This post `belongs_to :author` but for the model to know that this author actually exists in the `User` model, we will tell `belongs_to :author, class_name: 'User'`. and on the `User` model's side we will set the association like this `has_many :posts, foreign_key: 'author_id'`

- [ ] On the other two associations, the foreign_key will not be valid as it's already obvious that the `Post` has a unique `id` column in the database table

Reply:
Hello @MahmoudBakr23 
Wouldn't I need to change my schema in order to assign the `foreign_key` to `author_id`? The way the schema is presently constructed there is no `author_id` foreign key in any table. They are all called `users_id`. If that is the case, I don't see why I should make all those changes when I could just remove the `foreign_key` from the file and set up the associations with `User` instead of `Author`. Thank you for your time and let me know if you have any questions about the comment.

Reply:
I could also set `User` to `Author` but I would still need to reference the `users_id` foreign key. I find it a little unnecessary to change my schema in such a way when all I could do is remove the `foreign_keys` and call `User` as `Author` while still referencing the foreign key as `users_id`.

Reply:
I am also getting this error:
<img width="1057" alt="Screen Shot 2022-03-28 at 20 59 20" src="https://user-images.githubusercontent.com/82426682/160517322-393e08f5-542f-49ca-9d96-67833c440cc6.png">
With this model:
<img width="423" alt="Screen Shot 2022-03-28 at 21 00 07" src="https://user-images.githubusercontent.com/82426682/160517405-172738da-34b7-440a-88e9-14a4fe30382d.png">
Whereas the database updates itself with no issues with this model
<img width="589" alt="Screen Shot 2022-03-28 at 21 03 11" src="https://user-images.githubusercontent.com/82426682/160517747-d3a7ab75-2077-47f3-ad3d-374b69a270f1.png">
Could you please tell me if I set up anything in my project incorrectly? Because it seems like I do need the foreign_key in both `User` and `Post` for it to work as intended.

Reply:
@vechicin Exactly, and as the project requirements dictate, please add the foreign keys to your migrations correctly to stop receiving any errors.

You can first rollback to the original version of the scheme by `rails db:migrate VERSION=0`
Then modify your migrations and add the correct foreign keys, then migrate again so that the scheme gets updated.